File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Prompt user for some text
    string text = get_string("Text: ");

    // Count the number of letters, words, and sentences in the text
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    // Compute the Coleman-Liau index

    // Print the grade level
    printf("\n");

}

int count_letters(string text)
{
    // Return the number of letters in text
    // isalpha()
    int count = 0;
    for (int i = 0, j = strlen(text); i < j; i++)
    {
        if (isalpha(text[i]))
        {
            count ++;
        }
    }
    return count;
}

int count_words(string text)
{
    // Return the number of words in text
}

int count_sentences(string text)
{
    // Return the number of sentences in text
}

File: PSET_6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    # Prompt user for some text
    txt = get_string("Text: ")

    # Compute total number of letters, words, and sentences in the text
    letters = count_letters(txt)
    words = count_words(txt)
    sentences = count_sentences(txt)

    # Compute the Coleman-Liau index
    L = 100.0 * letters / words
    S = 100.0 * sentences / words
    index = 0.0588 * L - 0.296 * S - 15.8

    # Print the grade level
    if round(index) >= 16:
        print("Grade 16+")
    elif round(index) < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {int(round(index))}")

def count_letters(txt):
    return len(txt)


def count_words(txt):
    words = txt.split()
    return len(words)


def count_sentences(txt):
    dot_count = txt.count('.')
    excl_count = txt.count('!')
    question_count = txt.count('?')

    cs = dot_count + excl_count + question_count

    return cs


main()
